pkg/pxc/pxc.go

```diff
@@ -21,27 +21,31 @@ const appName = "pxc"
 
 var NoProxyDetectedError = errors.New("can't detect enabled proxy, please enable HAProxy or ProxySQL")
 
-func GetPrimaryPodDNSName(ctx context.Context, cl client.Client, cr *api.PerconaXtraDBCluster) (string, error) {
-	primary, err := GetPrimaryPod(ctx, cl, cr)
-	if err != nil {
-		return "", errors.Wrap(err, "get primary pod")
-	}
-	pxcSet := statefulset.NewNode(cr)
-	podList := corev1.PodList{}
-	if err := cl.List(ctx, &podList, &client.ListOptions{
+func getFirstReadyPodFQDN(ctx context.Context, cl client.Client, cr *api.PerconaXtraDBCluster) (string, error) {
+	node := statefulset.NewNode(cr)
+	sts := node.StatefulSet()
+
+	podList := new(corev1.PodList)
+	if err := cl.List(ctx, podList, &client.ListOptions{
 		Namespace:     cr.Namespace,
-		LabelSelector: labels.SelectorFromSet(pxcSet.Labels()),
+		LabelSelector: labels.SelectorFromSet(node.Labels()),
 	}); err != nil {
 		return "", errors.Wrap(err, "get pod list")
 	}
-	pxcSts := pxcSet.StatefulSet()
+
+	readyPods := make([]corev1.Pod, 0)
 	for _, pod := range podList.Items {
-		if pod.Status.PodIP == primary || pod.Name == primary {
-			primary = fmt.Sprintf("%s.%s.%s", pod.Name, pxcSts.GetName(), pxcSts.GetNamespace())
-			break
+		if k8s.IsPodReady(pod) {
+			readyPods = append(readyPods, pod)
 		}
 	}
-	return primary, nil
+	if len(readyPods) == 0 {
+		return "", errors.New("no ready pxc pods")
+	}
+	if len(readyPods) != int(cr.Spec.PXC.Size) {
+		return "", errors.New("waiting for pxc resize")
```

Comment:
IMO `waiting for pxc resize` is a bit confusing. maybe `waiting for all pxc pods to be ready`?